src/go/k8s/config/crd/bases/cluster.redpanda.com_redpandas.yaml

```diff
@@ -10438,10 +10527,19 @@ spec:
                               resource.
                             type: object
                           namespaceSelector:
-                            additionalProperties:
-                              type: string
                             description: Adds custom namespaceSelector to monitoring
                               resources
+                            properties:
+                              any:
+                                description: |-
+                                  Boolean describing whether all namespaces are selected in contrast to a
+                                  list restricting them.
+                                type: boolean
+                              matchNames:
+                                description: List of namespace names to select from.
+                                items:
+                                  type: string
+                                type: array
```

Comment:
So, I'm kind of concerned about this change given the fact that I'm thinking it's making the `namespaceSelector` field more restrictive... from an open map to a map with only two valid properties, `any` or `matchNames`. My guess is that the API server will potentially bomb if you have any existing objects with a `namespaceSelector` field that has any keys other than `any` or `matchNames` in it.

Any chance we could confirm and also whether we need this change to actually be backwards compatible? I think we could potentially do something like add some catch-all field to the underlying struct used to generate these CRDs if necessary.